nuagekubemon/client/nuageetcdclient.go

```diff
@@ -0,0 +1,358 @@
+/*
+###########################################################################
+#
+#   Filename:           nuageetcdclient.go
+#
+#   Author:             Siva Teja, Areti
+#   Created:            August 2, 2017
+#
+#   Description:        NuageKubeMon etcd Client Interface
+#
+###########################################################################
+#
+#              Copyright (c) 2017 Nuage Networks
+#
+###########################################################################
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"github.com/coreos/etcd/clientv3"
+	conc "github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/golang/glog"
+	"github.com/nuagenetworks/nuage-kubernetes/nuagekubemon/api"
+	"github.com/nuagenetworks/nuage-kubernetes/nuagekubemon/config"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	ETCD_BASE_PATH       = "/nuage.io/v1.0/"
```

Comment:
used for versioning.. makes more sense in a client server model.. currently it is not adding much... maybe useful in future if we want to change something in the tree structure without breaking the old version.